spirv/autogen_spirv.rs

```diff
@@ -2990,6 +3073,60 @@ impl core::str::FromStr for StoreCacheControl {
         }
     }
 }
+#[doc = "SPIR-V operand kind: [NamedMaximumNumberOfRegisters](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_named_maximum_number_of_registers_a_named_maximum_number_of_registers)"]
+#[repr(u32)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize))]
+#[cfg_attr(feature = "deserialize", derive(serde::Deserialize))]
+#[allow(clippy::upper_case_acronyms)]
+pub enum NamedMaximumNumberOfRegisters {
+    AutoINTEL = 0u32,
+}
+impl NamedMaximumNumberOfRegisters {
+    pub fn from_u32(n: u32) -> Option<Self> {
+        Some(match n {
+            0u32 => unsafe { core::mem::transmute::<u32, NamedMaximumNumberOfRegisters>(0u32) },
+            _ => return None,
+        })
+    }
+}
+#[allow(non_upper_case_globals)]
+impl NamedMaximumNumberOfRegisters {}
+impl core::str::FromStr for NamedMaximumNumberOfRegisters {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "AutoINTEL" => Ok(Self::AutoINTEL),
+            _ => Err(()),
+        }
+    }
+}
+#[doc = "SPIR-V operand kind: [FPEncoding](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_fp_encoding_a_fp_encoding)"]
+#[repr(u32)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize))]
+#[cfg_attr(feature = "deserialize", derive(serde::Deserialize))]
+#[allow(clippy::upper_case_acronyms)]
+pub enum FPEncoding {
+    Max = 0x7fffffff,
+}
+impl FPEncoding {
+    pub fn from_u32(n: u32) -> Option<Self> {
+        Some(match n {
+            _ => return None,
+        })
+    }
+}
+#[allow(non_upper_case_globals)]
+impl FPEncoding {}
+impl core::str::FromStr for FPEncoding {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            _ => Err(()),
+        }
+    }
+}
```

Comment:
This is a new optional enum operand for [`OpTypeFloat`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpTypeFloat) that's not yet defined to have any values/variants:

https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#FP_Encoding

I made some changes/hacks to `autogen` to support this variant-less enum, but need more to remove these empty `match` statements or decide if it should be an opaque enum (newtype struct on `u32`?) altogether depending on how enum values will be defined in the future.